test/specs/lib/reactComponentUtils-test.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react'
+import { getComponentName, areTypeNamesEqual } from 'src/lib/reactComponentUtils'
+import { createMockComponent } from 'test/utils'
+
+describe('getComponentName', () => {
+  it('returns null for invalid arguments', () => {
+    const invalidArgs = [undefined, null, '', {}]
+    invalidArgs.map(invalidArg => expect(getComponentName(invalidArg as any)).toBeFalsy())
+  })
+
+  it('returns the string component itself', () => {
+    const stringArgs = ['div', 'span', 'MyComponent']
```

Comment:
nit: would suggest to remove `MyComponent` from list, as is it a bit misleading: React doesn't support custom string components. Would be easier to read test cases if those will correspond to domain we are primarily interested in (while I do understand that you've tried to make tests generic):

```js
describe('getComponentName', () => {
  it('returns displayName for class component types', ...)
  it('returns function name for functional component types', ...)
  it('returns type itself for string component types', ...)

  it('returns null for invalid arguments', ...)
})
```

Reply:
N.A. anymore since the implementation was drastically simplified following decision:
https://github.com/stardust-ui/react/pull/496#discussion_r238708722